docs/proposals/20220725-managed-kubernetes.md

```diff
@@ -0,0 +1,521 @@
+
+For our example, GCP offers Node Pools that will manage the lifecycle of a pool machines that can scale up and down. We can use this service to implement machine pools:
+
+```go
+type GCPManagedMachinePoolSpec struct {
+// Location specifies where the nodes should be created.
+Location []string `json:"location"`
+
+// The Kubernetes version for the node group.
+Version string `json:"version"`
+
+// MinNodeCount is the minimum number of nodes for one location.
+MinNodeCount int `json:"minNodeCount"`
+
+// MaxNodeCount is the maximum number of nodes for one location.
+MaxNodeCount int `json:"minNodeCount"`
+
+...
+}
+```
+
+### Provider Implementers Documentation
+
+Its recommended that changes are made to the [Provider Implementers documentation](https://cluster-api.sigs.k8s.io/developer/providers/cluster-infrastructure.html) based on the recommending approach for representing managed Kubernetes in Cluster API. 
+
+Some of the areas of change (this is not an exhaustive list):
+
+- A new "implementing managed kubernetes" guide that contains details about how to represent a managed Kubernetes service in CAPI. The content will be based on option 3 from this proposal along with other considerations such as managed node and addon management.
+- Update the [Provider contracts documentation](https://cluster-api.sigs.k8s.io/developer/providers/contracts.html) to state that the same kind should not be used to satisfy 2 different provider contracts.
+- Update the [Cluster Infrastructure documentation](https://cluster-api.sigs.k8s.io/developer/providers/cluster-infrastructure.html) to provide guidance on how to populate the `controlPlaneEndpoint` in the scenario where the control plane creates the api server load balancer. We should include sample code.
+- Update the [Control Plane Controller](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/control-plane.html) diagram for managed k8s services case. The Control Plane reconcile needs to start when `InfrastructureReady` is true.
+
+## Other Considerations for CAPI
+
+### ClusterClass support for MachinePool
+
+- MachinePool is an important feature for managed Kubernetes as it is preferred over MachineDeployment to fully utilize the native capabilities such as autoscaling, health-checking, zone balancing provided by cloud providers’ node groups.
+- AKS supports MachinePool based worker nodes only and ClusterClass support for MachinePool is required. See [issue](https://github.com/kubernetes-sigs/cluster-api/issues/5991)
+
+### clusterctl integration
+
+- `clusterctl` assumes a minimal set of providers (core, bootstrap, control plane, infra) is required to form a valid management cluster.Currently, it does not expect a single provider being many things at the same time.
+- EKS in CAPA has its own control plane provider and a bootstrap provider packaged in a single manager. Moving forward, it will be great to separate them out.
```

Comment:
```suggestion
- `clusterctl` assumes a minimal set of providers (core, bootstrap, control plane, infra) is required to form a valid management cluster. Currently, it does not expect a single provider being many things at the same time.
- EKS in CAPA has its own control plane provider and a bootstrap provider packaged in a single manager. Moving forward, it would be great to separate them out.
```

Reply:
Changed